vcx/libvcx/src/utils/libindy/payments.rs

```diff
@@ -169,26 +169,39 @@ pub fn list_addresses() -> Result<Vec<String>, u32> {
     Ok(serde_json::from_str(&addresses).or(Err(error::INVALID_JSON.code_num))?)
 }
 
+fn is_valid_address(address: &str, method: &str) -> bool {
+    let prefix = format!("pay:{}", method);
+
+    if address.starts_with(&prefix) {
+        return true
+    }
+
+    return false
+}
+
 pub fn get_wallet_token_info() -> Result<WalletInfo, u32> {
     trace!("get_wallet_token_info >>>");
 
     let addresses = list_addresses()?;
 
+    let method = settings::get_config_value(settings::CONFIG_PAYMENT_METHOD)?;
     let mut balance = 0;
     let mut wallet_info = Vec::new();
 
     for address in addresses.iter() {
-        let mut info = get_address_info(&address)?;
+        if is_valid_address(&address, &method) {
+            debug!("getting address info for {}", address);
+            let mut info = get_address_info(&address)?;
 
-        for utxo in info.utxo.iter() { balance += utxo.amount as u64; }
+            for utxo in info.utxo.iter() { balance += utxo.amount as u64; }
 
-        wallet_info.push(info);
+            wallet_info.push(info);
+        }
```

Comment:
@glowkey @hadleym I think it's useful to add something as `else` branch.
1) Should it be an error to throw from function as result?
2) If no, please add `warn` or `info` or `debug` log. I think it's should be debug+ level (not trace), but don't have enough context to suggest exact level...

Reply:
Good suggestion. I have added a warning log statement.